registry/auth.go

```diff
@@ -101,14 +101,28 @@ func loginV2(authConfig *registry.AuthConfig, endpoint APIEndpoint, userAgent st
 	return "", "", errors.Errorf("login attempt to %s failed with status: %d %s", endpointStr, resp.StatusCode, http.StatusText(resp.StatusCode))
 }
 
-func v2AuthHTTPClient(endpoint *url.URL, authTransport http.RoundTripper, modifiers []transport.RequestModifier, creds auth.CredentialStore, scopes []auth.Scope) (*http.Client, error) {
+func v2AuthHTTPClient(endpoint *url.URL, authTransport http.RoundTripper, modifiers []transport.RequestModifier, creds auth.CredentialStore, scopes []auth.Scope, s *Service) (*http.Client, error) {
 	challengeManager, err := PingV2Registry(endpoint, authTransport)
 	if err != nil {
 		return nil, err
 	}
 
+	hostname := challengeManager.GetHost()
+
+	var newAuthTransport = authTransport
+	if len(hostname) > 0 {
+		tlsConfig, err := NewTLSConfig(hostname, s.config.isSecureIndex(hostname))
+		logrus.Debugf("Loading TLS config for host %s", hostname)
+		if err != nil {
+			logrus.Errorf("TLS config not found for host %s", hostname)
```

Comment:
Is it fine to continue in this case? If yes, then it should be a `Warnf` instead.
If it's an error (critical failure that means we can't continue), we should return an error and stop execution.